### DISCUSSION TITLE
Is there a Deno equivalent of the `libnode` endeavor? (C interface for cdylib builds)

### DISCUSSION THREAD

**Discussion by josh-hemphill** [denoland/deno#15810] [General]
For what's going on in node land that I'm referring to see the [PR](https://github.com/nodejs/node/pull/43542) and [MVP](https://github.com/mmomtchev/libnode)

Knowing nothing about cdylib generation, I thought I could just compile `deno_core` or `deno_runtime` as `cdylib` and be able to use it externally, but found out it needs `C` interfaces for that to work.
Personally I'd much prefer to use Deno for embedding in other applications, but I can't find any info on anyone else creating linkable binaries for Deno.

An off-the-cuff idea I have is if this exists; I wonder if it would be possible to use with `deno compile` to also be able to generate linkable binaries from javascript code. 👀

**Comment by littledivy** [comment]
You can write your own C interface for `deno_core` or `deno_runtime`. Note that both `core/` and `runtime/` APIs move very quickly so its unlikely Deno will expose an official C interface anytime soon.

```rust
#[no_mangle]
pub extern "C" fn create_runtime() {
  deno_core::JsRuntime::new(...);
}
```
